authorizer-app-backend/src/main/java/org/radarbase/authorizer/service/OAuth1RestSourceAuthorizationService.kt

```diff
@@ -0,0 +1,183 @@
+        logger.info("Requesting access token..")
+
+        val tokens = this.requestToken(authConfig.tokenEndpoint, RestOauth1AccessToken(payload.oauth_token!!, payload.oauth_token_secret, payload.oauth_verifier), sourceType)
+        return tokens?.let { mapToOauth2(it, sourceType) }
+    }
+
+    override fun refreshToken(user: RestSourceUser): RestOauth2AccessToken? {
+        return user.accessToken?.let { RestOauth2AccessToken(it, user.refreshToken) }
+    }
+
+    override fun revokeToken(user: RestSourceUser): Boolean {
+        val authConfig = configMap[user.sourceType]
+                ?: throw HttpBadRequestException("client-config-not-found", "Cannot find client configurations for source-type ${user.sourceType}")
+        val req = createRequest("DELETE", authConfig.deregistrationEndpoint!!, RestOauth1AccessToken(user.accessToken!!, user.refreshToken), user.sourceType)
+
+        return httpClient.newCall(req).execute().use { response ->
+            when (response.code) {
+                200, 204 -> true
+                400, 401, 403 -> false
+                else -> throw HttpBadGatewayException("Cannot connect to ${authConfig.deregistrationEndpoint}: HTTP status ${response.code}")
+            }
+        }
+    }
+
+    override fun deRegisterUser(user: RestSourceUser): RestSourceUser {
+        revokeToken(user)
+        val userDTO = userMapper.fromEntity(user)
+        userDTO.isAuthorized = false
+        return userRepository.update(user, userDTO)
+    }
+
+    override fun getAuthorizationEndpointWithParams(sourceType: String, callBackUrl: String): String {
+        logger.info("Getting auth endpoint..")
+        val authConfig = configMap[sourceType]
+                ?: throw HttpBadRequestException("client-config-not-found", "Cannot find client configurations for source-type $sourceType")
+
+        val tokens = this.requestToken(authConfig.preAuthorizationEndpoint, RestOauth1AccessToken(""), sourceType)
+        val params = mapOf<String, String?>(
+            OAUTH_ACCESS_TOKEN to tokens?.token,
+            OAUTH_ACCESS_TOKEN_SECRET to tokens?.tokenSecret,
+            OAUTH_CALLBACK to callBackUrl
+        )
+
+        return Url(authConfig.authorizationEndpoint, params).getUrl()
+    }
+
+    fun requestToken(tokenEndpoint: String?, tokens: RestOauth1AccessToken, sourceType: String): RestOauth1AccessToken? {
+        val req = createRequest("POST", tokenEndpoint.orEmpty(), tokens, sourceType)
+
+        return httpClient.newCall(req).execute().use { response ->
+            when (response.code) {
+                200 -> response.body?.string()
+                        ?.let { tokenReader.readValue<RestOauth1AccessToken>(parseParams(it)) }
+                        ?: throw HttpBadGatewayException("Service did not provide a result")
+                400, 401, 403 ->{
+                    logger.info(response.body?.string())
```

Comment:
I don't think the body should be logged, or only with debug log level scope with a proper message.